### ISSUE TITLE
Does not work in wine

### ISSUE THREAD

**Issue by nerdnils** [ClusterM/hakchi2#533]
hakchi2 does not work in wine.

"
WINEPREFIX="$HOME/wineprefix32" wine hakchi2/hakchi.exe

Unhandled Exception:
System.IO.FileNotFoundException: Could not load file or assembly 'System.Deployment, Version=2.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies.
File name: 'System.Deployment, Version=2.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'
[ERROR] FATAL UNHANDLED EXCEPTION: System.IO.FileNotFoundException: Could not load file or assembly 'System.Deployment, Version=2.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies.
File name: 'System.Deployment, Version=2.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'
"

Any chance to get it to work on linux?

**Comment by timehlers** [comment]
This is because you did not install .NET framework 3.5. I have installed it with winetricks, but there seems to be a config problem in winetricks. With a fresh ".wine" (rm -rf ~/.wine), when you tick dotnet35 it first tries to install dotnet2, which is afterwards incompatible with 3.5.

It took me a few hours to track that down and finding a workaround. You start winetricks, install dotnet40 and after that tick dotnet35.

Then hakchi2 works in general, but I have no idea how to passthrough the USB connection to install the kernel (and later the games)... If you find that out, please comment here.

Thanks

Tim

**Comment by bbsan2k** [comment]
Another approach would be to run the installer on wine and then run hakchi from your wine prefix folder.
Besides this: On Linux you can map the USB port to a wine USB port. On mac this does not seem to be possible sadly :(

**Comment by timehlers** [comment]
Yes, but this would still not install needed .NET 3.5. About the USB port: This is interesting! Do you have a link or little example? Thanks

**Comment by bbsan2k** [comment]
https://forum.ubuntuusers.de/topic/wine-wie-serielle-schnittstelle-auswaehlen/ <-- This is just an example. I am not even sure on how this actually works practically, as I do not have a GUI linux box set up right now!

**Comment by timehlers** [comment]
Thanks, but this only passes a serial-device to wine as comXXX. From what I understand is, that we need the raw USB-device, since allwinner sunxi is not plain serial. We would need to pass /dev/bus/usb/XXX/XXX to wine as whatever device, I don't know.

Actually I don't find any hint anywhere, only the statement from ClusterM, that linux users should use wine... No explanation how.

Maybe I don't get a central point. If anybody knows more, please comment on this here.

Thanks

**Comment by e8hffff** [comment]
USB should work straight up.  You may need to unblock UDEV with a rules set on some systems.  Like mentioned above check your dosdevices links in "cd ~/.wine/dosdevices" so they are pointing to ttySnn ports.  You maybe be able to auto build these in winecfg.

**Comment by timehlers** [comment]
This is basically not true. Did you try it? We need raw USB, which is not implemented in wine. Additionally wine has no direct access to any device, even storage must be mounted and passed as mountpoint, since direct block device access is not supported.

Second solution could be to use mono instead of wine, since it is a .NET-software. I have tried it, but as written from the OP, there are some functions unimplemented in mono (the OPs error message is from windows-mono in wine, but linux-mono throws the same).

I also tried [clovershell-client](https://github.com/ClusterM/clovershell-client), which indeed starts with mono and finds the SNES-mini! But communication fails after a few seconds. I assume, that there are timing issues. So even with the simple clovershell client there is no solution. I think of starting an issue there, too. Maybe the timing could be fixed more easily. Then we would have at least a daily usable solution and only need to flash the kernel once on windows.

**Comment by xswqaz** [comment]
Same error here

`Unhandled Exception:
System.IO.FileNotFoundException: Could not load file or assembly 'System.Deployment, Version=2.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies.
File name: 'System.Deployment, Version=2.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'
[ERROR] FATAL UNHANDLED EXCEPTION: System.IO.FileNotFoundException: Could not load file or assembly 'System.Deployment, Version=2.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies.
File name: 'System.Deployment, Version=2.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'
`

**Comment by Drakonas** [comment]
If someone wants to code a client for linux they have the right, but I suspect Cluster does not have the time to develop for two platforms.

**Comment by nerdnils** [comment]
Yes, he doesn't need to. But the FAQ says it runs in wine when in fact it doesn't I would not have reported this issue if Cluster had not said it does run in wine ;)

**Comment by Drakonas** [comment]
Oh does it? I'm sorry. I can make an adjustment on that.

**Comment by ClusterM** [comment]
Seems like WINE has not System.Deployment assembly. This assembly is used for auto-update. I'll try to make some workaround.

**Comment by cjunnior019** [comment]
Have you tried using virtualbox instead? I do, but when connecting Snes to computer the device is not recognized. Any idea?
